### PR TITLE
docs(adr): ADR-0038 engine-generated recovery placement as a requirement

### PR DESCRIPTION
**Pull request by Szczepanov**
Stacked on #453.

## Summary

- **Adds ADR-0038** (`Proposed`, documentation only) to make engine-generated recovery placement an explicit product-policy requirement instead of an emergent side effect of projected fatigue.
- **Why now:** #453 correctly projects the effective `activeDose`, exposing that the existing `restOrRecoveryDayCount > 0` simulation check was never owned by an explicit rule.
- **Proposed v1 policy:** target one **qualifying recovery date** in each seven consecutive local calendar dates, with at most six consecutive non-recovery dates when placement is feasible. This is explicitly a **product heuristic**, not a universal physiological rest prescription.
- **Impact:** no runtime behavior changes in this PR. Implementation remains a separate, policy-versioned stack.

## Reviewed contract

1. **Plan-less exact identity authority.** When no active authored `recovery_or_rest` requirement exists, use `EVERGREEN_GENERAL_COVERAGE_SET` + `general` only as recovery identity authority; do not fabricate an authored `PlanDefinition`/block.
2. **Explicit deadline.** A qualifying recovery on local date `R` gives `dueByDate = R + 7`; the due date is tier 1 before candidate selection. `WeeklyCoverageRequirement.windowEnd` is not this deadline.
3. **Explicit bootstrap boundary.** With unknown history and durable bootstrap `B`, `B` is only a deadline reference, **not recovery credit**. The first deadline is `B + 7`, the first evaluable window is `B + 1 ... B + 7`, and the pre-bootstrap unknown prefix is excluded. A real recovery before/on the first deadline replaces `B` as the reference and subsequent deadlines use `R + 7`.
4. **Durable bootstrap persistence.** `B` must be athlete-identity-scoped or deterministically derived from durable identity-scoped data; it must survive recomputation, restart and a second device/session.
5. **One recovery credit per local date.** Multiple qualifying identities on the same date still advance the policy once.
6. **Exact identities only.** Baseline qualifying identities include `rest_complete_01`, `recovery_mobility_tissue_01`, `recovery_breathwork_01`, and `cycling_recovery_spin_01`; category similarity alone does not qualify, and recovery spin still does not earn `aerobic_volume`.
7. **Historical Rest requires positive truth.** A recommendation alone or missing telemetry cannot manufacture completed Rest; generated Rest requires persisted decision/audit identity plus authoritative post-day occurrence reconciliation.
8. **ADR-0035 authored Rest retains provenance.** The bridge must preserve plan/revision/hash/directive/resolved-date identity and suppress credit on explicit override.
9. **ADR-0018 remains authoritative.** Recovery does not become a `RequiredRoleOccurrence`, never escalates to tier 0, and remains subordinate to D-SUPPORT—including the sole-ranked-candidate edge case.
10. **Best-effort, observable target.** A saturated horizon can pass `dueByDate` without recovery if higher-authority work/constraints win. That is not silently treated as success: a typed recovery miss/defer diagnostic is required.
11. **Evaluation must match the contract.** Simulation must use exact recovery identity, date-deduplicated credit, bootstrap-prefix semantics, maximum non-recovery streak and explicit miss/defer reporting. The coarse category-based `restOrRecoveryDayCount` remains only a smoke metric.

## Evidence boundary

The evidence review supports load/recovery management and individualization, but not a universal fixed weekly rest cadence. Therefore the exact `1 / 7 local dates` scalar remains a registered product heuristic (`evidenceCertainty: not_applicable`) bounded by `recovery.training.stress_recovery_balance`, not an evidence-derived physiological law.

## Relationship to existing ADRs

- **ADR-0016:** exact programming-role identity remains separate from broad category similarity.
- **ADR-0018:** no recovery reservation; deadline escalation remains subordinate to required-role viability.
- **ADR-0033:** runtime implementation requires Sports Knowledge Registry lineage, inventory/alignment tests and policy versioning.
- **ADR-0035:** authored Rest remains plan intent with replayable directive provenance.
- **ADR-0036:** day-level intraday recovery aggregation remains deferred.

## Validation / current status

- Documentation-only; no runtime code or persistence schema change.
- Branch is **ahead 9 / behind 0** of stacked #453 and GitHub reports the PR mergeable.
- All existing CodeRabbit inline threads are resolved.
- Current CodeRabbit status is green on head `957f0265499d2082536df64afc0682ad460ea7e7`.
- #457 has been synchronized to this exact head and now contains the same `B + 7` / `B + 1 ... B + 7` bootstrap contract.

## What implementation must prove

Before activation, the follow-up stack must prove the `R + 7` and `B + 7` boundaries, exact/date-deduplicated recovery identity, stable bootstrap across restart/device boundaries, positive historical Rest reconciliation, ADR-0035 provenance/override behavior, single-candidate D-SUPPORT, explicit miss diagnostics, deterministic placeable-scenario rolling-window behavior, saturated-horizon behavior, knowledge alignment and the `POLICY_VERSION` activation boundary.

### CODE REVIEW

File: docs/adr/0038-engine-generated-recovery-placement.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+## Decision
+
+Adopt **Option D**.
+
+### Requirement model
+
+- A baseline recovery requirement is seeded for every athlete, independent of
+  `PlanDefinition`. Where a plan definition exists, the authored `recovery_or_rest`
+  requirement continues to be the authority and is not duplicated.
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply docs/adr/0038-engine-generated-recovery-placement.md]: Agreed and fixed.

You are right that `buildCoverageState` empties on `!block` and `!descriptor` too, not only on `!planDefinition` — an out-of-block date or an unresolvable descriptor would have suppressed the baseline requirement while no authored requirement existed to replace it.

The Requirement model now deduplicates against **an active authored `recovery_or_rest` requirement**, not against `PlanDefinition` presence, and the implementation notes say explicitly that there are several early returns rather than one and the baseline must survive all of them.

File: docs/adr/0038-engine-generated-recovery-placement.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+- The requirement uses the existing `WeeklyCoverageRequirement` shape — a rolling
+  seven-day window and a minimum session count — so it participates in the ranking path
+  already built rather than introducing a parallel one.
+- Satisfying exposures are the ones that already carry the `recovery_or_rest` coverage key.
+  An ADR-0035 authored rest day satisfies it. A same-day athlete override that replaces
+  rest with training does not.
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply docs/adr/0038-engine-generated-recovery-placement.md]: Agreed and fixed — this was a genuine gap between two accepted ADRs.

Added a "How an ADR-0035-authored rest day satisfies it" section. The directive **materializes a `recovery_or_rest` coverage exposure** on its resolved plan-local date rather than being counted through a parallel path, which keeps ADR-0016's authored-identity rule intact and makes the requirement satisfied by construction on an authored rest date instead of by a special case in ranking.

Two consequences are now stated explicitly: an authored rest date cannot also attract generated recovery because the requirement is already met, and an ADR-0035 `athleteOverridesAuthoredRest` removes the materialized exposure along with the rest, so normal placement resumes. The exposure must carry the directive's provenance and must not be attributed to a template the plan did not author.

Also took the hyphenation nit.

File: docs/adr/0038-engine-generated-recovery-placement.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+### Ranking authority
+
+- An unmet recovery minimum ranks at deferred-support tier 2 while the window has slack,
+  exactly as today. Recovery does not compete with hard roles by default.
+- When the days remaining in the window are no more than the still-unmet minimum, the
+  requirement escalates to **tier 1**. This mirrors the existing overdue-hard-role repair
+  pattern in `coverageNeedTierForTemplate`.
+- It never escalates to tier 0. Tier 0 is date-level programming authority reserved for a
+  nominated anchor, and recovery must not be able to steal an authored anchor date.
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply docs/adr/0038-engine-generated-recovery-placement.md]: Addressed, though the conclusion is different from the one suggested — ADR-0018 settled it more cleanly than expected.

Added an "Allocation, viability and miss reporting" section:

**Reservations.** ADR-0018 already rules that "`recovery_or_rest` remains satisfied by the existing coverage ledger and recovery policy; it does not reserve a discretionary training date." So the baseline requirement is *not* promoted to a `RequiredRoleOccurrence`, the `weeklyAllocation.ts:140` filter stays, and escalation is a ranking-order change only. Option C is now rejected outright rather than retained as a component — adopting it would mean overturning an accepted ADR, which this one does not propose.

**D-SUPPORT viability.** Already covered, and the escalated candidate is subordinate to it: "On a train/modify-tier unreserved date, a discretionary Rest selection consumes the date and therefore receives the same stateful viability proof as any supporting selection. It is rejected when it would remove the last proven required-role allocation." A tier-1 escalated recovery candidate is subject to that proof and may be rejected by it. In a true recover tier, ADR-0018 Rest-first already outranks the proof.

**Miss reporting.** Because recovery takes no reservation, D-MISS does not report it — an unmet requirement produces no `WeeklyRoleAllocationReport` outcome. That is consistent, but it makes an unfulfilled recovery requirement invisible, so the ADR now requires a separate diagnostic (window closed unfulfilled; D-SUPPORT rejected an escalated candidate) rather than forcing recovery into the reservation model to inherit D-MISS.

**Test.** Your plan-less scenario is now a required acceptance test.

The `buildCoverageState` half of the finding was already the ADR's first implementation note; it is now sharper about there being several early returns to change.

File: docs/adr/0038-engine-generated-recovery-placement.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+Adopt **Option D**.
+
+### Requirement model
+
+- A baseline recovery requirement is seeded for every athlete, independent of
+  `PlanDefinition`. Where a plan definition exists, the authored `recovery_or_rest`
+  requirement continues to be the authority and is not duplicated.
+- The requirement uses the existing `WeeklyCoverageRequirement` shape — a rolling
+  seven-day window and a minimum session count — so it participates in the ranking path
+  already built rather than introducing a parallel one.
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply docs/adr/0038-engine-generated-recovery-placement.md]: Partly accepted — reworded for precision, but not the underlying rule.

ADR-0011 §3 ("Anchors nudge; they do not command") scopes itself explicitly to its own three post-gate modifiers: "All three are ranking modifiers applied *after* the hard gates — Patches 4 and 5 are multipliers, Patch 6 is a tie-break." Those are the ×1.35 role boost, the ×0.3 adjacency suppression and the variety tie-break.

The coverage-need tier is a different, later mechanism — introduced by ADR-0016/ADR-0018, implemented in `coverageNeedTierForTemplate`, and a lexicographic sort key rather than a multiplier. So there is no contradiction to resolve: both statements are true of their own mechanism, and this ADR changes neither.

My wording invited the confusion by not naming the mechanism, so that is fixed: the tier-0 statement is now attributed to the ADR-0016 coverage-need ordering, and says recovery must not displace a nominated anchor *through the coverage tier*. I dropped "authored anchor date".

One thing worth keeping from this: the `coverage.ts` comment calling tier 0 "date-level programming authority" — which is where my phrasing came from — is loose wording for a ranking key, and reading it as a global claim about anchors is a reasonable mistake to make. The ADR now flags it for re-scoping when that function is next touched. I did not change the comment here, since this PR is documentation-only and that is an engine file.

**Comment by Szczepanov** [comment]
Completed a deeper architecture + evidence review and pushed the resulting ADR hardening onto this PR.

### What changed in this review

The main direction remains **Option D: plan-independent recovery placement with deadline escalation**, but the ADR now closes several contracts that were underspecified enough to make a follow-up implementation incorrect even if it appeared to satisfy the existing scenario test:

- **Plan-less identity authority:** a synthetic `WeeklyCoverageRequirement` alone is inert because `coverageNeedTierForTemplate()` needs phase + descriptor identity. The ADR now requires a recovery-only product-policy identity context backed by `EVERGREEN_GENERAL_COVERAGE_SET` / `general`, without fabricating an authored plan.
- **Real deadline semantics:** the policy is now explicitly one qualifying recovery exposure in every seven consecutive local dates (max six consecutive non-recovery dates). A latest recovery on `R` means `R + 7` is the next due date; the existing trailing counter cannot stand in for that deadline.
- **Stable bootstrap / unknown history:** missing history is not proof of no recovery, and a fresh rolling recomputation may not keep sliding the deadline forward.
- **Exact recovery identities:** category similarity is insufficient under ADR-0016. Baseline identity authority must at least cover `rest_complete_01`, `recovery_mobility_tissue_01`, `recovery_breathwork_01`, and `cycling_recovery_spin_01`. The recovery spin remains separate from `aerobic_volume` credit.
- **Projected vs historical Rest:** a Rest recommendation can satisfy the current forecast but cannot become completed recovery next morning merely because no activity was recorded. Generated complete Rest needs a positive/auditable day-outcome path reconciled against performed training.
- **ADR-0035 bridge:** an authored `restDays` directive needs a recovery fact with plan/revision/content-hash/directive/date provenance; the existing generic coverage source is not sufficient by itself. Athlete override removes/suppresses that credit.
- **D-SUPPORT edge case:** escalation remains unreserved under ADR-0018 and must stay subordinate to required-role viability, including when recovery is the only ranked candidate.
- **Acceptance tests:** `restOrRecoveryDayCount > 0` is retained only as a smoke check. The ADR now requires an actual rolling seven-date invariant, boundary-day, bootstrap, exact-identity, authored-rest, contradictory-training, D-SUPPORT, miss-diagnostic, and descriptor-consistency tests.

### Evidence boundary

I also re-checked the external recovery literature. It supports deliberate stress/recovery management and individualization, but does **not** justify presenting a universal weekly full-rest day as physiology. The ADR therefore uses **qualifying recovery exposure** and explicitly labels the `1 / 7 dates` scalar as product policy (`heuristic`, `evidenceCertainty: not_applicable`), bounded by the scientific recovery claim rather than derived from it.

Sources checked:
- Kellmann et al. 2018 recovery consensus: https://pubmed.ncbi.nlm.nih.gov/29345524/
- Li et al. 2024 endurance recovery umbrella review: https://pubmed.ncbi.nlm.nih.gov/38753045/
- Sandbakk et al. 2025 Norwegian world-class endurance coaching study: https://pubmed.ncbi.nlm.nih.gov/40278987/

### Repository status after the push

- Current head: `45f5652e96571874d3be900cf3aa42455b902e41`
- PR is **mergeable** against stacked base #453; no conflict reported.
- `docs/README.md` diff was re-checked after the edits and contains only the intended ADR-0038 index addition.
- All four existing CodeRabbit inline threads are resolved; no unresolved inline thread remains.
- GitHub reports **no Actions workflow run on the current head**, so I am not claiming a CI-green result. This is still documentation-only and intentionally does not change runtime policy.

My review conclusion: the ADR is now substantially more implementation-ready and evidence-safe, while correctly remaining **Proposed**. The next runtime PR should implement the acceptance contract + knowledge lineage + `POLICY_VERSION` change together rather than treating the four current scenario failures as simple baseline drift.

File: docs/adr/0038-engine-generated-recovery-placement.md [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+requirements below.
+
+### Product-policy scalar
+
+For v1:
+
+* minimum: **one qualifying recovery exposure**;
+* interval: **every seven consecutive local calendar dates**;
+* equivalent invariant: there may be at most **six consecutive non-recovery dates** once
+  the policy has authoritative history or a stable bootstrap epoch;
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply docs/adr/0038-engine-generated-recovery-placement.md]: Valid — fixed in 235e9162.

The scalar section now states the credit unit explicitly: **one qualifying local date, not one qualifying exposure**, with the reasoning that two measures coexist. `WeeklyCoverageRequirement` counts `completedSessions + projectedSessions`, which is exposure-based, while the deadline anchors on the latest qualifying *date*. A Rest and a mobility exposure sharing a date must yield one credit.

At the v1 minimum of one this changes no behavior, which is precisely why it was worth writing down — it stops being inert the moment a later policy version raises the minimum, and a latent double-count is a poor thing to leave behind that change.

Added acceptance test 12: a date carrying two qualifying identities advances the policy by one credit, not two.

File: docs/adr/0038-engine-generated-recovery-placement.md [hunk 6]
```diff
@@ -0,0 +1,428 @@
...
+
+Absence of a recorded workout is not proof that the athlete rested, and absence of a
+recorded recovery exposure is not automatically proof that the athlete did not recover.
+The implementation must distinguish **authoritative no-recovery history** from **unknown
+history**.
+
+A brand-new or history-incomplete athlete must use a stable bootstrap epoch for the policy;
+the deadline may not slide forward every time the rolling plan is recomputed. The exact
+storage/source of that epoch is an implementation choice, but the acceptance test must
+prove that repeated daily recomputation cannot postpone recovery indefinitely.
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply docs/adr/0038-engine-generated-recovery-placement.md]: Valid — fixed in 235e9162.

"The exact storage/source of that epoch is an implementation choice" was too permissive, and you named why: a process-local value, an in-memory default, or anything recomputed from the current date resets on restart and differs on a second device, and each reset moves the deadline forward. That is the indefinite postponement the section exists to prevent, and no single-run test would catch it.

The epoch must now be **durable and identity-scoped** — persisted against the athlete identity, or derived deterministically from data that already is. Acceptance test 4 is extended to prove stability across a restart and across a second device/session, not merely across repeated recomputation within one run.

File: docs/adr/0038-engine-generated-recovery-placement.md [hunk 7]
```diff
@@ -0,0 +1,428 @@
...
+* recovery escalated;
+* D-SUPPORT rejected escalated recovery;
+* recovery deferred because of a hard safety/availability constraint;
+* historical recovery state unknown rather than unmet.
+
+### Ranking authority
+
+* With slack before `dueByDate`, qualifying recovery remains deferred-support **tier 2**.
+* On `dueByDate`, or when already overdue, qualifying recovery becomes **tier 1**.
+* It never escalates to tier 0.
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply docs/adr/0038-engine-generated-recovery-placement.md:316]: Valid, and the sharpest of the three — fixed in 235e9162.

The ADR asserted "at most six consecutive non-recovery dates" as an invariant while also making escalation tier-1-only, non-reserving and subordinate to D-SUPPORT. Those cannot both hold. A horizon saturated with tier-0 or required-role work can pass `dueByDate` with nothing placed.

Resolved as **best-effort under hard constraints**, which is the reading consistent with ADR-0018:

- the six-date rule is now stated as a target the policy pursues, not an invariant it can guarantee;
- Ranking authority says plainly that escalation neither reserves a date nor outranks tier 0, and that a missed deadline is a reported diagnostic rather than a silent absence;
- acceptance test 1 is scoped to scenarios where recovery is actually placeable, since an unscoped assertion would test a guarantee the policy does not make;
- acceptance test 11 is the saturated-horizon case you asked for: the deadline passes unfulfilled, nothing is forced through a protected role, and the miss surfaces on the diagnostic.

The ADR also now records the alternative explicitly: a hard guarantee would require recovery to reserve a date, which means amending ADR-0018. That remains out of scope here, but it is stated as a decision available rather than left implicit.

**Comment by Szczepanov** [comment]
Aligned ADR-0038 with the implementation-plan bootstrap contract identified during #457 review.

Changes in `957f0265`:

- unknown-history bootstrap date `B` is now explicitly a **deadline reference only**, never recovery credit;
- first deadline is `B + 7` local calendar days;
- first evaluable seven-date window is `B + 1 ... B + 7`;
- the pre-bootstrap unknown prefix is excluded from success/failure evaluation;
- a real qualifying recovery before/on the first deadline becomes the new `R` and subsequent deadlines use `R + 7`;
- acceptance tests now require those boundary semantics in addition to durable identity-scoped persistence across recomputation/restart/device boundaries.

I also refreshed the PR description so it no longer calls the seven-date rule a hard invariant and instead matches the reviewed best-effort/diagnostic contract.

#457 has been merged forward to this exact head; its stack is now ahead 5 / behind 0 and only its two intended documentation files differ.
